week-1-practise/level-3/magic-8-ball.js

```diff
@@ -0,0 +1,222 @@
+      eightBall = 'Cannot predict now'
+      break
+    case 14:
+      eightBall = 'Concentrate and ask again'
+      break
+    case 15:
+      eightBall = "Don't count on it"
+      break
+    case 16:
+      eightBall = 'My reply is no'
+      break
+    case 17:
+      eightBall = 'My sources say no'
+      break
+    case 18:
+      eightBall = 'Outlook not so good'
+      break
+    case 19:
+      eightBall = 'Very doubtful'
+      break
+  }
+  return eightBall
+}
+
+// The answer should come from shaking the ball
+let answer
+
+// When checking the answer, we should tell someone if the answer is
+// - very positive
+// - positive
+// - negative
+// - very negative
+function checkAnswer(message) {
+  if (message == 'It is certain') {
+    answer = 'Very positive'
+  } else if (message == 'It is decidedly so') {
+    answer = 'very positive'
+  } else if (message == 'Without a doubt') {
+    answer = 'very positive'
+  } else if (message == 'Yes - definitely') {
+    answer = 'very positive'
+  } else if (message == 'You may rely on it') {
+    answer = 'very positive'
+  } else if (message == 'As I see it, yes') {
+    answer = 'positive'
+  } else if (message == 'Most likely') {
+    answer = 'positive'
+  } else if (message == 'Outlook good') {
+    answer = 'positive'
+  } else if (message == 'Yes') {
+    answer = 'positive'
+  } else if (message == 'Signs point to yes') {
+    answer = 'positive'
+  } else if (message == 'Reply hazy, try again') {
+    answer = 'very negative'
+  } else if (message == 'Ask again later') {
+    answer = 'very negative'
+  } else if (message == 'Better not tell you now') {
+    answer = 'very negative'
+  } else if (message == 'Cannot predict now') {
+    answer = 'very negative'
+  } else if (message == 'Concentrate and ask again') {
+    answer = 'very negative'
+  } else if (message == "Don't count on it") {
+    answer = 'negative'
+  } else if (message == 'My reply is no') {
+    answer = 'negative'
+  } else if (message == 'My sources say no') {
+    answer = 'negative'
+  } else if (message == 'Outlook not so good') {
+    answer = 'negative'
+  } else {
+    answer = 'negative'
+  }
+
+  return answer
+}
+
+/* ======= TESTS - DO NOT MODIFY ===== */
+const log = console.log
+let logged
+console.log = function() {
+  log(...arguments)
+  logged = arguments[0]
+}
+
+function test(test_name, expr) {
+  let status
+  if (expr) {
+    status = 'PASSED'
+  } else {
+    status = 'FAILED'
+  }
+
+  logged = undefined
+  console.log(`${test_name}: ${status}`)
+}
+
+const validAnswers = []
+function testAll() {
+  const answer = shakeBall()
+  test(
+    `shakeBall logs "The ball has shaken!"`,
+    logged === 'The ball has shaken!',
+  )
+  test(`shakeBall returns an string answer"`, typeof answer === 'string')
+  test(
+    `checkAnswer returns the level of positivity"`,
+    ['very positive', 'positive', 'negative', 'very negative'].includes(
+      checkAnswer(answer),
+    ),
+  )
+}
+
+testAll()
```

Comment:
Works great! I see you ended up not using that `eightballValid` array, I think in the next couple of classes we'll learn how to take advantage of using arrays to make exercises like this look even nicer 😎